domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -2108,7 +2108,7 @@ private FacetManager buildFacetManager(CQLWrapper wrapper, QueryHelper queryHelp
     List<FacetField> facets, Handler<AsyncResult<Results<T>>> replyHandler) {
 
     get(conn, table, clazz, DEFAULT_JSONB_FIELD_NAME, filter, returnCount,
-      false, setId, facets, replyHandler);
+      true, setId, facets, replyHandler);
```

Comment:
Can you explain why this is required? returnIdField=true is deprecated since RMB 25.0.0 (RMB-346) because the id field is always copied into jsonb->'id'. The client should use jsonb->'id' and not that id internal database column.

Reply:
Remember that this is Criterion, not CQL .. I've updated the issue https://issues.folio.org/browse/RMB-532
with more description

Reply:
I should of course make a test case.. But this is not done for the 29.1.1 release where the single line change was made.

Reply:
Regressions like this illustrates that it is problematic with multiple "query" languages.. Really we should have CQL only IMHO

Reply:
I've fixed the real cause: setId=true.
And reverted this hack.